src/main/java/com/bekzat/gym/service/map/UserService.java

```diff
@@ -0,0 +1,48 @@
+package com.bekzat.gym.service.map;
+
+import com.bekzat.gym.dao.UserRepository;
+import com.bekzat.gym.dto.CredentialsDto;
+import com.bekzat.gym.dto.UserReadDto;
+import com.bekzat.gym.exceptions.AuthenticationException;
+import com.bekzat.gym.exceptions.UserNotFoundException;
+import com.bekzat.gym.mapper.UserMapper;
+import com.bekzat.gym.model.User;
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+import java.util.Optional;
+
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class UserService {
+
+    private final UserRepository userRepository;
+    private final UserCredentialsService userCredentialsService;
+    private final UserMapper userMapper;
+
+
+    public UserReadDto findById(Long id, CredentialsDto credentialsDto) {
+        if (userCredentialsService.checkCredentials(credentialsDto)) {
+            throw new AuthenticationException("Invalid credentials");
+        }
+        log.info("Finding user by ID: {}", id);
+        return userRepository.findById(id)
+                .map(userMapper::toDto)
+                .orElseThrow(() -> new UserNotFoundException(id.toString()));
```

Comment:
it is not good manner to use toString method everywhere you want - please use Long.toString(value)